### PR TITLE
Fix Case Sensitive F12 for Functions

### LINKED ISSUE #571: F12 for non-class function should not be case sensitive
image

Expected: We go to the function definition

### PR DESCRIPTION
**Pull request by MrayerMSFT**
Should fix #571. @HuanhuanSunMSFT @AndreSayreMSFT

### CODE REVIEW

File: PowerShellTools/LanguageService/NavigationExtensions.cs [hunk 1]
```diff
@@ -114,7 +114,7 @@ private static IEnumerable<FunctionDefinitionAst> FindDefinition(CommandAst refe
             {
                 // If in the same scope as the reference call, the function must be defined before the call
                 var definitions = scope.Statements.OfType<FunctionDefinitionAst>().
-                    Where(def => def.Name == reference.GetCommandName() && def.Extent.EndOffset <= reference.Extent.StartOffset);
+                    Where(def => def.Name.ToLower() == reference.GetCommandName().ToLower() && def.Extent.EndOffset <= reference.Extent.StartOffset);
```
Comment (ASayre) [inline PowerShellTools/LanguageService/NavigationExtensions.cs]: Use standard string comparison methods, not ToLower

**Comment by HuanhuanSunMSFT** [comment]
Looks good now.

File: PowerShellTools/LanguageService/NavigationExtensions.cs [hunk 2]
```diff
@@ -114,7 +114,7 @@ private static IEnumerable<FunctionDefinitionAst> FindDefinition(CommandAst refe
             {
                 // If in the same scope as the reference call, the function must be defined before the call
                 var definitions = scope.Statements.OfType<FunctionDefinitionAst>().
-                    Where(def => def.Name == reference.GetCommandName() && def.Extent.EndOffset <= reference.Extent.StartOffset);
+                    Where(def => def.Name.Equals(reference.GetCommandName(), System.StringComparison.OrdinalIgnoreCase) && def.Extent.EndOffset <= reference.Extent.StartOffset);
```
Comment (ASayre) [inline PowerShellTools/LanguageService/NavigationExtensions.cs]: nit: using System;

**Comment by ASayre** [comment]
looks good, sans Nit  & subsequent AppVeyor happiness

File: PowerShellTools/LanguageService/NavigationExtensions.cs (hunk 2 above)
Comment (MrayerMSFT) [inline PowerShellTools/LanguageService/NavigationExtensions.cs]: Ahhhh I didn't even notice that. Womp womp. Will fix. >.<

**Comment by MrayerMSFT** [comment]
Alright, nit has been addressed and AppVeyor looks happy. Going to merge now. Closes #571

**Pull request merged**